crates/lair_keystore_api/src/sodium_secretstream/crypt.rs

```diff
@@ -71,21 +72,32 @@ pub(crate) struct PrivCryptRecv(BoxStream<'static, LairResult<Box<[u8]>>>);
 
 impl PrivCryptRecv {
     /// Initialize the new decryption receiver.
-    pub fn new(recv: PrivFramedRecv, dec: sss::SecretStreamDecrypt) -> Self {
+    pub fn new(
+        recv: PrivFramedRecv,
+        dec: sodoken::secretstream::State,
+    ) -> Self {
         let recv = futures::stream::try_unfold(
             (recv, dec),
             |(mut recv, mut dec)| async move {
                 let cipher = match recv.next().await {
                     None => return Ok(None),
                     Some(cipher) => cipher?,
                 };
-                let cipher = sodoken::BufRead::from(cipher);
-                let msg =
-                    sodoken::BufWrite::new_no_lock(cipher.len() - sss::ABYTES);
-                dec.pull(cipher, <Option<sodoken::BufRead>>::None, msg.clone())
-                    .await?;
-                let msg = msg.try_unwrap().unwrap();
-                Ok(Some((msg, (recv, dec))))
+                let mut cipher = sodoken::LockedArray::from(cipher);
```

Comment:
```suggestion
```

cipher doesn't need to be locked

Reply:
https://github.com/holochain/lair/pull/143/commits/b6f64ac499f6c25e33b6b125e4b3e59e90761dca